### DISCUSSION TITLE
Free/Libre DB instead of MongoDB

### DISCUSSION THREAD

**Discussion by KenMacD** [danny-avila/LibreChat#3985] [Feature Requests & Suggestions]
It would be nice if this ran on a free DB. I haven't tried it with FerretDB yet but requiring MongoDB is a bit of a surprise for a 'libre' project.

**Comment by danny-avila** [comment]
The decision to use MongoDB was a technical one. The name was voted by the community surrounding LibreChat pre-naming circa March 2023, and is attributed to the “libre” nature of the project itself, the core frontend and backend source code which is fully open source and MIT licensed. 

As for MongoDB, I am happy to swap it out for a drop in replacement, as is possible with some databases already, but a complete refactor for something else is not in order. MongoDB is easily distributed and open enough to know its source code, which makes it widely available and reliable. Not least of all, it does allow flexibility as a NoSQL database with constantly changing data structures especially as AI services multiply and evolve.

**Reply by KenMacD** [reply]
I understand (and thanks for the quick reply). I'm happy to hear that using MongoDB isn't a rigid dependency. Maybe I'll try FerretDB and see if it works.

I get that MongoDB is 'available', I just found it surprising because other 'libre' projects tend to have a pretty strict view on free vs non-free. I was looking to create a NixOS package to try out the project, but the reliance on MongoDB adds at least an hour of compiling to build the project because NixOS (like many distros) doesn't package it.

I guess if I just wanted you to know so that future development towards other DBs could be considered.

**Reply by AlekSi** [reply]
Hi folks, FerretDB maintainer here. LibreChat works with FerretDB; we're writing a blog post about it.
One benefit of using FerretDB instead of MongoDB is that we support vector search (https://docs.ferretdb.io/guides/vector-search/), so you will need FerretDB and PostgreSQL only.

**Reply by gwozdzm** [reply]
The blog post about using LibreChat + FerretDB is available:
https://blog.ferretdb.io/replacing-mongodb-with-ferretdb-librechat/

**Reply by ion-elgreco** [reply]
Maybe propose a PR to change the default
